### PR TITLE
Add test cases for args and building

### PR DESCRIPTION
**Pull request by charbull**
Refactor to test args and others

### CODE REVIEW

**Review by berkoben — APPROVED** [review]

File: tools/validators/instance_validator/tests/handler_test.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import os
+
+from absl.testing import absltest
+from validate import handler
+
+_TESTCASE_PATH = os.path.join('.', 'tests', 'fake_instances')
+
+class HandlerTest(absltest.TestCase):
+
+  def testOneBuildingExist(self):
```
Comment (nkilmer) [inline tools/validators/instance_validator/tests/handler_test.py:30]: Could you add the name of the methods under test to these test names?

File: tools/validators/instance_validator/tests/handler_test.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      input_file = os.path.join(_TESTCASE_PATH, 'GOOD',
+                                'good_building_type.yaml')
+      args = ['--input', input_file]
+      instance_handler = handler.ValidationHelper(args)
+      instance_handler.Validate()
+    except SyntaxError:
+      self.fail('ValidationHelper:Validate raised ExceptionType unexpectedly!')
+
+  def testOneBuildingExistFails(self):
+    with self.assertRaises(Exception):
```
Comment (nkilmer) [inline tools/validators/instance_validator/tests/handler_test.py:41]: Which specific exception is expected to be raised?

File: tools/validators/instance_validator/validate/handler.py [hunk 3]
```diff
@@ -136,7 +137,7 @@ def StartTelemetryValidation(self, subscription, service_account_file,
       sub.Listen(validator.ValidateMessage)
 
 
-  def _PrepareArgs(self):
+  def ParseArgs(self, args):
```
[the remark this answers is not in the document]
Comment (charbull) [reply tools/validators/instance_validator/validate/handler.py]: done

File: tools/validators/instance_validator/tests/handler_test.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      # there is missing building type in the test file
+      input_file = os.path.join(_TESTCASE_PATH, 'GOOD', 'good_links.yaml')
+      args = ['--input', input_file]
+      instance_handler = handler.ValidationHelper(args)
+      instance_handler.Validate()
+
+  def testTelemetryArgs(self):
```
Comment (nkilmer) [inline tools/validators/instance_validator/tests/handler_test.py:48]: It's unclear to me what about the telemetry args is being tested here.

File: tools/validators/instance_validator/tests/handler_test.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    try:
+      input_file = os.path.join(_TESTCASE_PATH, 'GOOD',
+                                'good_building_type.yaml')
+      args = ['--input', input_file, '--service-account', 'file',
+              '--subscription', 'some-subscription']
+      handler.ValidationHelper(args)
+    except SystemExit:
+      self.fail('ValidationHelper:Validate raised ExceptionType unexpectedly!')
+
+  def testTelemetryArgsMissingSubscription(self):
```
Comment (nkilmer) [inline tools/validators/instance_validator/tests/handler_test.py:58]: What about missing service account?

**Review by nkilmer — CHANGES_REQUESTED** [review]
See comments in files.

File: tools/validators/instance_validator/tests/handler_test.py (hunk 1 above)
Comment (charbull) [reply tools/validators/instance_validator/tests/handler_test.py]: done

File: tools/validators/instance_validator/tests/handler_test.py (hunk 2 above)
Comment (charbull) [reply tools/validators/instance_validator/tests/handler_test.py]: done

File: tools/validators/instance_validator/tests/handler_test.py (hunk 4 above)
Comment (charbull) [reply tools/validators/instance_validator/tests/handler_test.py:48]: updated

File: tools/validators/instance_validator/tests/handler_test.py (hunk 5 above)
Comment (charbull) [reply tools/validators/instance_validator/tests/handler_test.py:58]: updated

**Review by nkilmer — APPROVED** [review]

**Pull request merged**